### PR TITLE
added authentication for new user

### PR DESCRIPTION
**Pull request by sanchit203**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -37,6 +49,7 @@ app.use((req, res, next) => {
 // Routes
 app.use("/", indexRoute)
 app.use("/listings", listingRoute)
+app.use("/", userRoute)
```
Comment (sanchit203) [inline app.js]: two routes on same api prefix? indexRoute and userRoute both on "/" ?? we can give this route as "auth"

File: models/user.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+    email: {
+        type: String,
+        required: true
+    }
+});
+
+userSchema.plugin(passportLocalMongoose);
```
Comment (sanchit203) [inline models/user.js]: can you add a comment for why we need this?

File: routes/userRoute.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+userRoute.get("/signup", (req, res) => {
+    res.render("users/signup.ejs");
+});
+
+userRoute.post("/signup", wrapAsync(async (req, res) => {
+    try {
+        let { username, email, password } = req.body;
+        const newUser = new User({ email, username });
+        const registeredUser = await User.register(newUser, password);
+        console.log({ registeredUser });
```
Comment (sanchit203) [inline routes/userRoute.js]: will this console log the password as well? 
If yes, thn won't this be a security concern

File: app.js (hunk 1 above)
Comment (jstsanam) [reply app.js]: done

File: models/user.js (hunk 2 above)
Comment (jstsanam) [reply models/user.js]: done

File: routes/userRoute.js (hunk 3 above)
Comment (jstsanam) [reply routes/userRoute.js]: changed the consoles

**Review by jstsanam — APPROVED** [review]

**Pull request merged**